app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt

```diff
@@ -278,14 +278,21 @@ class BrowserFragment : EngineViewLifecycleFragment(), Session.Observer {
         activity?.menuInflater?.inflate(R.menu.menu_context_hometile, menu)
     }
 
+    private fun exitFullScreenAndBack() {
+        // Backing while full-screened can lead to unstable behavior (see #1224),
+        // so we always attempt to exit full-screen before backing
+        requireComponents.sessionManager.getEngineSession()?.exitFullScreenMode()
+        if (session.canGoBack) requireComponents.sessionUseCases.goBack.invoke()
```

Comment:
Interesting. So if the user perform `NavigationEvent.BACK` but nothing in the back stack, it will always exitFullScreenMode(). But not for `onBackPressed()`?

Reply:
When we hit the `browserOverlay.isVisible && !isUrlEqualToHomepage` branch of `onBackPressed`, we close the overlay and go back to the content without changing the URL.  I think in this case we should probably keep the video full-screened.  I'm not super happy about how many similar code paths we have here though.